Sources/Sentry/SentryEnvelope.m

```diff
@@ -9,10 +11,21 @@ @implementation SentryEnvelopeHeader
 
 // id can be null if no event in the envelope or attachment related to event
 - (instancetype)initWithId:(NSString *_Nullable)eventId
+{
+    SentrySdkInfo *sdkInfo = [[SentrySdkInfo alloc] initWithSdkName:SentryMeta.sdkName
+                                                   andVersionString:SentryMeta.versionString];
+    self = [self initWithId:eventId andSdkInfo:sdkInfo];
+
+    return self;
+}
```

Comment:
H: @bruno-garcia was in favor of not having an overload for it, but instead we'd pass null to `andSdkInfo` in the other method, so if one uses it, they won't mistake as they know `sdkInfo` is needed, but it's nullable if not available

Reply:
I like the default here, because otherwise we would have to deal with `SdkInfo` in many places. If we keep it like this as a user of `SentryEnvelope` I don't care how to properly create the `SdkInfo`.

Reply:
not sure if I follow it, the idea is to force users to pass a `sdkInfo`, it's fine to pass null to keep retro compatibility, but you need to think that this is a mandatory field, so offering an overload that skips it is not a good thing.

Reply:
So do you suggest to remove the `_Nullable` for `andSdkInterface` to make it mandatory?

Reply:
this class changed now and this comment is out of sync with the source code, but what I'd do is:
`andSdkInterface` has a nullable field, so it's possible to pass null for retro compatibility.
I'd remove a method that doesn't require a valid `andSdkInterface` or null`, so it doesn't look like optional.

`create(NonNull eventId, Nullable sdkInterface)`

no other overload.

Reply:
I would not remove `initWithId:eventId`, because it sets the `SdkInfo` for you. You don't have to deal with the complexity of creating `SdkInfo` manually. `initWithId:eventId` is for me the recommended to init a `SentryEnvelopeHeader`